File: 2week/남규/3. Minimum Time Difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        vector<int> points;
        for(auto timePoint : timePoints){
            int h1 = timePoint[0] - '0';
            int h2 = timePoint[1] - '0';
            int m1 = timePoint[3] - '0';
            int m2 = timePoint[4] - '0';
            
            int v = m2 + m1 * 10 + h2 * 60 + h1 * 60 * 10;
            points.push_back(v);
        }
        
        sort(points.begin(), points.end());
        int ans = INT_MAX;
        
        for(int i = 0; i < points.size() - 1; i++){
            int diff = points[i + 1] - points[i];
            ans = min(ans, diff);
        }
        return min(ans, points[0] + 1440 - points.back());
    }
};

File: 2week/근우/homework/3.minimum-time-difference.py
from typing import List

LIMIT = 24 * 60
MAX = LIMIT


class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        timePoints = list(map(lambda x: int(x[:2]) * 60 + int(x[3:]), timePoints))
        timePoints.sort()
        timePoints.append(timePoints[0] + LIMIT)
        ans = MAX
        for i in range(len(timePoints) - 1):
            ans = min(ans, timePoints[i + 1] - timePoints[i])
        return ans


print(Solution().findMinDifference(timePoints=["23:59", "00:00"]))
